pilot/pkg/proxy/envoy/v2/discovery.go

```diff
@@ -110,23 +114,41 @@ func (s *DiscoveryServer) Register(rpcs *grpc.Server) {
 // Singleton, refresh the cache - may not be needed if events work properly, just a failsafe
 // ( will be removed after change detection is implemented, to double check all changes are
 // captured)
-func periodicRefresh() {
+func (s *DiscoveryServer) periodicRefresh() {
 	ticker := time.NewTicker(periodicRefreshDuration)
 	defer ticker.Stop()
 	for range ticker.C {
-		PushAll()
+		adsLog.Infof("ADS: periodic push")
+		AdsPushAll(s)
 	}
 }
 
-// PushAll implements old style invalidation, generated when any rule or endpoint changes.
-// Primary code path is from v1 discoveryService.clearCache(), which is added as a handler
-// to the model ConfigStorageCache and Controller.
-func PushAll() {
-	versionMutex.Lock()
-	version = time.Now()
-	versionMutex.Unlock()
+// Push metrics are updated periodically (10s default)
+func (s *DiscoveryServer) periodicRefreshMetrics() {
+	envOverride := os.Getenv("V2_METRICS")
```

Comment:
why can't this be a CLI option? that would make things easy.
Also, PILOT_METRICS_PUSH_INTERVAL is better than V2_METRICS